### DISCUSSION TITLE
Easier emptying of the Find Box (Search Box)

### DISCUSSION THREAD

**Discussion by Peter-202122** [sumatrapdfreader/sumatrapdf#4001] [Ideas]
The necessity to delete manually the last search term typed into the Find Box (before typing the next one) is not very comfortable; especially when you have to perform many searches.

So my suggestion is easy to describe:

An "X" in the right corner of the Find Box so that by clicking on it the last search term is removed automatically. Just as it works e.g. in the search box here at Github.  

Thanks for considering such an addition.

**Comment by GitHubRulesOK** [comment]
agreed it is not intuitive but try 
double click next word (strike a phrase) you wish to search will transfer that word/phrase into search using `/` key this will include all characters in the area selection. a different selection will replace the first using `/`
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/ccbf79a1-4ac3-4874-84a2-a0f999c0f420)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/a153164f-e7c2-49e5-bb13-a620e5dc3d8c)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/d5d8b1d8-7ec4-4c54-be98-9f50380bb826)

**BE AWARE (beware)**
while the search box is highlighted (**may be done via a single click**, OR CTRL+A) then the next key will replace that entry (so that will auto clear and provide 1st character for area search.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/85321095-154e-4b1b-aa4f-267d45433fa0)

**Comment by Peter-202122** [comment]
Thanks for your reply.

The first procedure (the one you describe above the "Be Aware") is quite complicated. ;-)

The other one (under the "Be aware") is not bad but it requires that first you click somewhere into the document (= focus outside the search box) and afterwards you click again in the search box itself. Otherwise (= if you stay with the cursor in the search box) the highlighting via a single click does not work and you have to delete the old search term manually. 

Personally I often search for a word and having found it I would like to search immediately for the next one. In this case the necessary intermediate step (= mouse click into the document) plus going back into the search box is cumbersome and slows down the work flow. So at least in this case an emptying by clicking on a "X" would be much easier.

**Reply by GitHubRulesOK** [reply]
Agreed not intuitive (Find is find single characters only) but I will often when the cursor is in the box at end of line **(without a space it is double click same as select one word)** or overstrike the part area to alter, so just as fast as move to an x and click it
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/51c7990f-e7ef-42a4-aafc-86757e45a449)

**Comment by Peter-202122** [comment]
I am not quite sure what you mean but I assume it is the following:

When the focus (cursor) is within the find box, I can achieve the highlighting of the existing entry by doubleclicking (with the cursor position after the last character). So I can afterwards overwrite this entry as described by you above. I just tried it and it works. 

But as you write, it is not very intuitive. - I have no idea how simple or difficult would be the implementation of a solution like the "X". But if it is easy to realize, I would clearly prefer such a straightforward solution.

**Reply by giwul70** [reply]
Agree a [X] button would slightly be faster. That said, as an example only, say I would be searching for the word "comm" in some document. The cursor is still in the find box (incremental search). Whilst hitting the ">"  button I skip thru the document. When done, I might do "ctrl-a" and enter new search immy (as the cursor is still in the find box), or ctrl-a followed by backspace.

**Comment by maxbruening** [comment]
Possibly the same/related:  consecutive searches by highlighting text and then pressing "ctrl-f"  do not work perfectly.  After a first search if one directly starts another search by highlighting another string of text and presses "ctrl-f",  the new string appears in the search box, but in the pdf the old (or next?) instance of the old search term is highlighted.

**Comment by endolith** [comment]
It used to be that pressing Ctrl+F would highlight the search box and if I start typing `banana` it replaces the text in the search box and immediately starts searching for `banan` but now if the box contains `apple` and I press Ctrl+F and start typing `banana` the text is appended *afterwards*, searching for `applebanana` which is not what I want.

**Reply by kjk** [reply]
other people want other behavior; I can't please everyone
